### PR TITLE
Add inset box-shadow to images

### PR DESCRIPTION
**Pull request by davemart-in**
## Description

I went through and added inset box-shadows to as many images as I could. The reader feed should be fully compatible now. 

Here's a close up of what you should be seeing (especially around light areas of the picture border):

CleanShot 2022-10-13 at 10 41 43

## Known issues

We still have some things to sort out with individual images on full post views. I'm setting the box-shadow on the `figure` tag, which works great most of the time, except when a caption is included, it's a child of figure, leading to this unwanted white space:

image

Unfortunately you can't set the inset box-shadow on the image itself, it has to be applied one layer up.

There is also no CSS class on `figure` to indicate which images have a caption and which ones don't.

I wonder if we could convince core to add a `has-caption` class to `figure`? Or if there some way to add one via filters?

## Index of known updates

| **Location** | **Variation**         | **Rounded Corners** | **Inset border** |
|--------------|-----------------------|---------------------|------------------|
| Feed         | Single image          | ✅                   | ✅                |
| Feed         | Zoom in image         | ✅                   | ✅                |
| Feed         | 2 image               | ✅                   | ✅                |
| Feed         | 3 image               | ✅                   | ✅                |
| Feed         | 4 image               | ✅                   | ✅                |
| Feed         | Gallery               | ✅                   | ✅                |
| Feed         | Video                 | ✅                   | ✅                |
| Feed         | Recommended posts     | ✅                   | ✅                |
| Full post    | Featured image        | ✅                   | ✅                |
| Full post    | Gallery               | ✅                   | ✅                |
| Full post    | More In (Recommended) | ✅                   | ✅                |
| Full post    | Single image          | ✅                   |                  |

## Related

#68987

### CODE REVIEW

File: client/blocks/reader-full-post/_content.scss [hunk 1]
```diff
@@ -58,6 +58,30 @@
 
 .is-reader-page .reader-full-post__story-content {
 	padding-top: 0 !important;
+
+	.wp-block-gallery .wp-block-image {
```
Comment (creativecoder) [inline client/blocks/reader-full-post/_content.scss:68]: You probably need to add classes for images added with the classic editor and/or with the Classic block as well?

**Review by creativecoder — COMMENTED** [review]
Looks like there might be a regression here, post with 2 images in the feed, right side border radius seems to be missing (testing in Firefox):

| **Before** | **After** |
| - | - |
| image | image |

**Comment by davemart-in** [comment]
> Looks like there might be a regression here, post with 2 images in the feed, right side border radius seems to be missing (testing in Firefox):
> 
> **Before**	**After**
> image	image

@creativecoder this regression should be fixed now. I also added an index of updates in the description above for easier reference while testing:

CleanShot 2022-10-13 at 13 52 46

**Comment by eoigal** [comment]
I have seen images in search and related posts no longer get expanded, not sure where in PR though that is responsible.

### With PR
Screenshot 2022-10-14 at 12 00 08
Screenshot 2022-10-14 at 11 59 29

### Without PR

Screenshot 2022-10-14 at 12 01 32

UPDATE: this looks related to https://github.com/Automattic/wp-calypso/issues/68994

**Comment by eoigal** [comment]
> I wonder if we could convince core to add a has-caption class to figure? Or if there some way to add one via filters?

I was thinking about this. Do you think it would help if we did a regex replace on all images in `post.content` when displaying the full view content in reader. We use the regex to add container divs (or some markup) to each img tag, these divs could have a class that we use to style with inset box shadow. Do you think that would be worth pursuing?

File: client/blocks/reader-full-post/_content.scss (hunk 1 above)
Comment (davemart-in) [reply client/blocks/reader-full-post/_content.scss:68]: Hmm... We probably are going to have to deal with images in a classic block the same way we're dealing with images with a caption (by injecting a container element around the image). There is nothing here that tells me this image is in a classic block, and I can't attach a box-shadow to a generic `p` tag as that will surely lead to edge case issues down the line. cc @eoigal 

Here's the code for an image in a classic block:

CleanShot 2022-10-14 at 10 54 56

**Comment by davemart-in** [comment]
> I have seen images in search and related posts no longer get expanded, not sure where in PR though that is responsible.
> UPDATE: this looks related to https://github.com/Automattic/wp-calypso/issues/68994

Yeah, let's deal with this in a separate PR @eoigal. My thought was that we could compare the width to the height of an image pre-rendering and if the width is equal to the height (square) or a vertical image that we could apply a CSS class that would allow us to adjust accordingly (targeting max-width, vs. max-height).

**Comment by davemart-in** [comment]
> I was thinking about this. Do you think it would help if we did a regex replace on all images in post.content when displaying the full view content in reader. We use the regex to add container divs (or some markup) to each img tag, these divs could have a class that we use to style with inset box shadow. Do you think that would be worth pursuing?

Let's definitely do this!

File: client/blocks/reader-full-post/_content.scss (hunk 1 above)
Comment (davemart-in) [reply client/blocks/reader-full-post/_content.scss:68]: For testing: `/read/feeds/133479798/posts/4320194408`

**Pull request merged**